app/models/grda_warehouse/ch_enrollment.rb

```diff
@@ -74,5 +74,324 @@ def self.update_existing!
+    # Lines 4, 10, 17, and 24 (3.917.3)
+    def self.approximate_start_date(enrollment, date: enrollment.EntryDate)
+      ch_start_date = [enrollment.DateToStreetESSH, enrollment.EntryDate].compact.min
+      project = enrollment.project
+      days = if date != enrollment.EntryDate && (project.so? || project.es? && project.bed_night_tracking?)
+        dates_in_enrollment_between(enrollment, enrollment.EntryDate, date).count + (enrollment.EntryDate - ch_start_date).to_i
+      else
+        (date - ch_start_date).to_i
+      end
+      result = days > 365 ? :yes : nil
+      { result: result, display_value: "#{days} days" }
+    end
+
+    THREE_OR_FEWER_TIMES_HOMELESS = [1, 2, 3].freeze
+    TWELVE_OR_MORE_MONTHS_HOMELESS = [112, 113].freeze # 112 = 12 months, 113 = 13+ months
+
+    # Lines 5, 11, 18, and 25 (3.917.4)
+    def self.num_times_homeless(enrollment)
+      value = enrollment.TimesHomelessPastThreeYears
+
+      result = if THREE_OR_FEWER_TIMES_HOMELESS.include?(value)
+        :no
+      elsif dk_or_r_or_missing(value)
+        dk_or_r_or_missing(value)
+      end
+
+      { result: result, display_value: value }
+    end
+
+    # Lines 6, 12, 19, and 26 (3.917.4)
+    def self.total_months_homeless(enrollment, date: enrollment.EntryDate)
+      value = enrollment.MonthsHomelessPastThreeYears
+      return { result: :yes, display_value: value - 100 } if TWELVE_OR_MORE_MONTHS_HOMELESS.include?(value)
+
+      # If you don't have time prior to entry, day calculation above will catch any days during the enrollment
+      # If you have time prior to entry and we are looking at an arbitrary date, we need to add
+      # the months served. (This is only used for Chronic-at-PIT calculation, not Chronic-at-Entry).
+      if date != enrollment.EntryDate && enrollment.MonthsHomelessPastThreeYears.present? && enrollment.MonthsHomelessPastThreeYears > 100
+        project = enrollment.project
+        months_in_enrollment = if project.so? || project.es? && project.bed_night_tracking?
+          dates_in_enrollment_between(enrollment, enrollment.EntryDate, date).map do |d|
+            [d.month, d.year]
+          end.uniq.count
+        else
+          month_count = (date.year * 12 + date.month) - (enrollment.EntryDate.year * 12 + enrollment.EntryDate.month)
+          # Subtract 1 from this number if the [project start date] does not fall on the first of the month.
+          month_count -= 1 if month_count.positive? && enrollment.EntryDate.day != 1
+          month_count
+        end
+        months_prior_to_enrollment = enrollment.MonthsHomelessPastThreeYears - 100
+        sum = months_prior_to_enrollment + months_in_enrollment
+        return { result: :yes, display_value: sum } if sum > 11
+      end
+
+      return { result: dk_or_r_or_missing(value), display_value: value } if dk_or_r_or_missing(value)
+
+      { result: nil, display_value: value - 100 }
+    end
+
+    # TODO: test boundaries days/months for entry/exit, NbN, and SO
```

Comment:
Is this still a todo?  Can you add a few tests if it is?

Reply:
I'm not totally sure what it meant, this TODO was already here. It looks to me like we are already testing everything here https://github.com/greenriver/hmis-warehouse/blob/de64b54fea450ba0e0f71eb714e2772fb8fbfa5f/spec/models/grda_warehouse/hud/enrollment_spec.rb#L83  I removed the note, but let me know if there are other tests you think need to be added.